packages/app/src/runner/index.ts

```diff
@@ -144,6 +144,22 @@ function teardownSpec () {
   return getEventManager().teardown(getMobxRunnerStore())
 }
 
+let isTorndown = false
+
+/**
+ * Called when navigating away from the runner page.
+ * This will teardown the reporter, event maanger, and
+ * any associated events.
+ */
+export function teardown () {
```

Comment:
Ensures we correctly teardown the reporter and event manager when we navigate away, ensuring a "clean slate".

Reply:
Can this function be made async and await the teardown and resetReporter? You mentioned some edge-cases, and `isTorndown` can be set to true before it's actually torndown unless we let the promises resolve.

Reply:
Sure, done.